src/components/Header.jsx

```diff
@@ -135,28 +152,50 @@ const Header = ({
           </>
         )}
       </Flex>
-      {showBackNavWarningModal && (
-        <GenericWarningModal
-          displayTitle="Warning"
-          displayText="You have unsaved changes. Are you sure you want to navigate away from this page?"
-          onProceed={toggleBackNav}
-          onCancel={() => setShowBackNavWarningModal(false)}
-          proceedText="Yes"
-          cancelText="No"
-        />
-      )}
-      {showStagingWarningModal && (
-        <GenericWarningModal
-          displayTitle=""
-          displayText="Your changes may take some time to be reflected. <br/> Refresh your staging site to see if your changes have been built."
-          displayImg="/publishModal.svg"
-          displayImgAlt="View Staging Modal Image"
-          onProceed={handleViewStaging}
-          onCancel={() => setShowStagingWarningModal(false)}
-          proceedText="Proceed to staging site"
-          cancelText="Cancel"
-        />
-      )}
+      <GenericWarningModal
+        isOpen={isWarningOpen}
+        onClose={onWarningClose}
+        displayTitle="Warning"
+        displayText="You have unsaved changes. Are you sure you want to navigate away from this page?"
+      >
+        <Button colorScheme="danger" onClick={onWarningClose}>
+          No
+        </Button>
+        <Button onClick={toggleBackNav}>Yes</Button>
+      </GenericWarningModal>
+      <Modal isOpen={isStagingOpen} onClose={onStagingClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader />
+          <ModalCloseButton />
+          <ModalBody>
+            <Text textStyle="body-2">
+              Your changes may take some time to be reflected. <br />
+              Refresh your staging site to see if your changes have been built.
+            </Text>
+          </ModalBody>
+          <ModalFooter>
+            <HStack w="100%" spacing={2} justifyContent="flex-end">
+              <Button colorScheme="danger" onClick={onStagingClose}>
+                Cancel
+              </Button>
+              <Skeleton isLoaded={!!stagingUrl}>
+                <Button
+                  as={Link}
+                  textDecoration="none"
+                  _hover={{
+                    textDecoration: "none",
+                    bgColor: "primary.600",
+                  }}
+                  href={stagingUrl}
+                >
+                  <Text color="white">Proceed to staging site</Text>
+                </Button>
+              </Skeleton>
+            </HStack>
+          </ModalFooter>
+        </ModalContent>
+      </Modal>
```

Comment:
couldn't we also use `GenericWarningModal` here?

Reply:
I left this out originally because of the image that was supposed to be added here, but if we allow for more flexible modal bodies, this makes sense to include here now - changed in https://github.com/isomerpages/isomercms-frontend/pull/941/commits/d209f907ac75192c25af93006168ec91711f3def but without the image - will be added after the PR inserting the image has been added